wayback/tests/test_client.py

```diff
@@ -514,3 +524,37 @@ def test_rate_limit_error_includes_retry_after(self, requests_mock):
             session.request('GET', 'http://test.com')
 
         assert excinfo.value.retry_after == 10
+
+    @mock.patch('requests.Session.send', side_effect=return_timeout)
+    def test_timeout_applied_session(self, mock_class):
+        # Is the timeout applied through the WaybackSession
+        session = WaybackSession(timeout=1)
+        res = session.request('GET', 'http://test.com')
+        assert res.text == '1'
+        # Overwriting the default in the requests method
+        res = session.request('GET', 'http://test.com', timeout=None)
+        assert res.text == 'None'
+        res = session.request('GET', 'http://test.com', timeout=2)
+        assert res.text == '2'
```

Comment:
This *is* hacky, but I think it works fine for now! I’m fine merging this. 👍

If you want to do something fancier, you could check out this timeout test in web-monitoring-diff: https://github.com/edgi-govdata-archiving/web-monitoring-diff/blob/2fe3088b04bb2a58fdb5fec1e92d4fed181233f3/web_monitoring_diff/tests/test_server_exc_handling.py#L192-L202

(It tests that a proxy server correctly times out while contacting the “upstream” server with the actual content. The `patch` decorator makes the proxy being tested have a short timeout, and `responder` and `SimpleHttpServer` create a quick little HTTP server to use as the upstream, which waits longer than the expected timeout before responding. That’s way more complex than what you’re doing here, though. It would be more complete, but not really necessary. Also note that package uses `async/await`, so the solution here couldn’t be exactly the same.)